### ISSUE TITLE
Should modify iisnode.yml when it is under a folder and pointed by package.json/scripts/start

### ISSUE THREAD

**Issue by compulim** [projectkudu/kudu#2411]
The location of `iisnode.yml` is based on the location of the Node.js entrypoint, but not `web.config`.

Currently, Kudu modify the `iisnode.yml` in the root, instead of locating it at the same level of the Node.js entrypoint.

### Repro steps.
1. Write a web server entrypoint named `lib/app.js` and it simply spill out `process.env.node_env`
2. Modify `package.json` and set `scripts.start` to `node lib/app.js`
3. Write a `lib/iisnode.yml` with content `node_env: under-lib`
4. Write another `iisnode.yml` with content `node_env: root`
5. Have a usual `web.config` that run the project with `iisnode`, for example, copy from https://gist.github.com/davidebbo/2571104
5. Deploy thru Project Kudu (in my case, it's Azure Web App)

#### Expected
1. `lib/iisnode.yml` is appended with pointer to Node.js of the latest version (`7.7.4` when I deployed to Azure)
2. When browse to the site, it say `under-lib`

#### Actual
1. `iisnode.yml` is modified instead
2. When browse to the site, it say `root`

### Project structures.
```
/iisnode.yml (this file should be ignored)
/package.json (this file has scripts.start pointing to /lib/app.js)
/lib/iisnode.yml (this file should be modified)
/lib/app.js (entrypoint and response with process.env.node_env)
```

### The log/error given by the failure.
The wrong `iisnode.yml` is modified.

### Mention any other details that might be useful.
I verified the behavior of locating `iisnode.yml` in an environment with IIS and iisnode, no Kudu and not on Azure.

You can take my [Docker image](https://hub.docker.com/r/compulim/iisnode) that have `iisnode` preinstalled and try it out yourself easily.

**Comment by davidebbo** [comment]
Can you please [share a minimal repo](https://github.com/projectkudu/kudu/wiki/Using-a-git-repo-to-report-an-issue) that demonstrates the problem? It makes it easier (and less error prone) than having to create one step by step.

**Comment by compulim** [comment]
@davidebbo Could you try out my repro at my [repository](https://github.com/compulim/repro-kudu-iisnode-yml)?

You will see there are two `iisnode.yml`, one under root, and one under `/lib/`. The one at root is a dummy and it is never used by iisnode. It set `NODE_ENV` to `root`.

> In my case, I use the Project Kudu on Azure Web App. You can use the "Deploy to Azure" from `README.md` for a quick deployment.

After you deploy it via Kudu, the site will print `{"NODE_ENV":"under-lib",...}`. That means iisnode is loading `/lib/iisnode.yml`, instead of `/iisnode.yml`.

But when you look at `/iisnode.yml`, you will see it get modified by Project Kudu:

```
node_env: root
nodeProcessCommandLine: "D:\Program Files (x86)\nodejs\7.7.4\node.exe"
```

And for the `/lib/iisnode.yml` that is actually loaded by iisnode, you won't see the change there. It just looks like:

```
node_env under-lib
```

It seems Project Kudu is modifying the incorrect `iisnode.yml`.

**Comment by compulim** [comment]
iisnode read the configuration file `iisnode.yml` based on the location of the current request, you can see the code [here](https://github.com/tjanczuk/iisnode/blob/c816a6740ae63e2c3065653539ea18b9c5edfce6/src/iisnode/cmoduleconfiguration.cpp#L924) and the [reference of `GetScriptTranslated`](https://msdn.microsoft.com/en-us/library/ms692661(v=vs.90).aspx).

That means, Project Kudu would need to lookup the `iisnode.yml` based on `web.config/system.webServer/handlers/add[name="iisnode"]/@path`.

**Comment by davidebbo** [comment]
I think the issue is that the logic that the [logic](https://github.com/projectkudu/kudu/blob/7e83ac73d33af63862190a0ef04200881a639287/Kudu.Core/Scripts/selectNodeVersion.js#L149) that deals with using the subpath is only applied when there is no web.config in the repo. Can you try deleting it and letting Kudu's web.config generation do its thing?

**Comment by davidebbo** [comment]
Possible Kudu fix: 0ef0cec5f6687a0e8e927fe7dbfbd19ae6fc9ca2. Better diff that ignores blanks: https://github.com/projectkudu/kudu/commit/0ef0cec5f6687a0e8e927fe7dbfbd19ae6fc9ca2?w=1

**Comment by jvano** [comment]
Hi

If the problem persists and is related to running it on Azure App Service, please open a support incident in Azure:
https://learn.microsoft.com/en-us/azure/azure-portal/supportability/how-to-create-azure-support-request

This way we can better track and assist you on this case

Thanks,

Joaquin Vano
Azure App Service
